src/database/PersonCategories/index.ts

```diff
@@ -0,0 +1,58 @@
+import dbConnection, { tableNames, execute } from '../connection';
+import logger from '@/utilities/modules/logger';
+
+const personCategoriesTable = () => dbConnection()(tableNames.PERSON_CATEGORIES);
+const personCategoryAttributesTable = () => dbConnection()(tableNames.PERSON_CATEGORY_ATTRIBUTES);
+
+interface PersonCategory {
+  personcategoryid: number;
+  personcategoryname: string;
+}
+
+interface PersonCategoriesAndAttributes {
+  [key: string]: string[];
+}
+
+const builders = {
+  /** QueryBuilder for getting all person categories */
+  personCategories() {
+    return personCategoriesTable().select('*');
+  },
+  /** QueryBuilder for getting a map of all person categories and their attributes */
+  personCategoryAttributes(categoryId: number) {
+    return personCategoryAttributesTable()
+      .join('personattributetypes', 'personcategoryattributes.attrid', 'personattributetypes.attrid')
```

Comment:
do you think it'd be a good idea to have constants for the names of tables and attributes somewhere? this can prevent easy typos and bugs

Reply:
I didn't want to because this was faster to dev and try things. I guess now that it's working it wouldn't be a bad idea.

The variable names end up being really long though since words get separates by underscores and the variable would be prepended with `tableName.`. So maybe not for now. And also they're only being used one place right now. I think it'll be much more useful when we start describing the database in the core instead of an external script.